<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <title>SEEKR</title>

    <script type="text/javascript" src="{{ url_for('static', filename='jquery-3.2.1.js') }}"></script>
    <script type="text/javascript" src="{{ url_for('static', filename='external/jquery/jquery.js') }}"></script>
    <script type="text/javascript" src="{{ url_for('static', filename='jquery-ui.js') }}"></script>
    <script src="https://cdn.rawgit.com/eligrey/canvas-toBlob.js/f1a01896135ab378aa5c0118eadd81da55e698d8/canvas-toBlob.js"></script>
    <script src="https://cdn.rawgit.com/eligrey/FileSaver.js/e9d941381475b5df8b7d7691013401e171014e89/FileSaver.min.js"></script>
    <script src="http://d3js.org/d3.v3.min.js"></script>

    <script type="text/javascript" src="{{ url_for('static', filename='savePNG.js') }}"></script>
    <script type="text/javascript" src="{{ url_for('static', filename='main.js') }}"></script>
    <!--<link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='main.css') }}">-->

    <!--<script type="text/javascript" src="../static/jquery-3.2.1.js"></script>-->
    <!--<script type="text/javascript" src="../static/main.js"></script>-->

    <link rel="stylesheet" type="text/css" href="../static/main.css">

</head>
<body>

<div id="main-tabs" class="row">

    <div class="menu">
        <div class="row">
            <div class="single_column container">

                <div class="nav_menu">
                    <a href="/"></a>
                    <!--<img class="nav_img" src="../static/UNC.jpg"/>-->
                    <img class="nav_img" src="{{ url_for('static', filename='UNC.jpg') }}">
                </div>

                <ul class="row">
                    <li class="tab nav_left"><a href="#main"><h2>Run SEEKR</h2></a></li>
                    <li class="tab tab2 nav_center"><a href="#visual"><h2>Results</h2></a></li>
                    <li class="tab tab3 nav_right"><a href="#help"><h2>Help</h2></a></li>
                </ul>

            </div>
        </div>
    </div>

    <!--------------------------------------------------------->
    <!--------------------------------------------------------->
    <div id="main" class="main">


        <div class="row">

            <div class="column">
                <h1>Upload user RNA set</h1>
            </div>

            <div class="column">
                <h1>Select or upload comparison RNA set</h1>
            </div>
        </div>

        <div class="row">

            <div id="input_set" class="column container">

                <div class="row">
                    <div class="dropzone workzone">

                        <form id="user_set_form" method="post" enctype="multipart/form-data">

                            <input type="file" name="file" id="user_set_files">

                            <h2>Drag and drop .fa file here or click to upload</h2>

                            <!--<img src="../static/dragndrop.svg">-->
                            <img src="{{ url_for('static', filename='dragndrop.svg') }}">

                            <h2 id="user_file_text"></h2>

                        </form>
                    </div>
                </div>

                <p id="user_warning" class="warning">Warning: This set contains too many sequences to visualize. <br>Files must contain < 200 sequences for visualization.</p>
            </div>


            <div id="comparison_set" class="column container">

                <div class="row">
                    <ul>
                        <li class="tab select_left"><a href="#select_file"><h2>Select set</h2></a></li>

                        <li class="tab tab2 select_right"><a href="#upload_file"><h2>Upload set</h2></a></li>

                    </ul>
                </div>



                <div id="select_file"class="row">
                    <div class="workzone_tabbed">
                        <div class="row">
                            <h2>Choose from standard sets</h2>
                        </div>

                        <select name="comparison_set_reference" id="comparison_set_reference">
                            <option value="gencode_human_set">All Human lncRNA (Gencode)</option>
                            <option value="gencode_mouse_set">All Mouse lncRNA (Gencode)</option>
                            <option value="user_set" selected="selected">User Set</option>
                        </select>
                    </div>
                </div>

                <div id="upload_file" class="row">
                    <div class="dropzone workzone_tabbed">
                        <form id="comparison_set_form" method="post" enctype="multipart/form-data">

                            <input type="file" name="file" id="comparison_set_files">

                            <h2>Drag and drop .fa file here or click to upload</h2>

                            <!--<img src="../static/dragndrop.svg">-->
                            <img src="{{ url_for('static', filename='dragndrop.svg') }}">

                            <h2 id="comparison_file_text"></h2>

                        </form>
                    </div>
                </div>

                <p id="comparison_warning" class="warning">Warning: This set contains too many sequences to visualize. <br>Files must contain < 200 sequences for visualization.</p>
            </div>
        </div>


        <div class="row break">

            <div class="single_column_small">
                <h1>SEEKR settings</h1>
            </div>
        </div>

        <div class="row">

            <div class="single_column_small container">

                <div class="row">

                    <div class="column">
                        <h2>Select normalization set</h2>
                    </div>

                    <div class="column">
                        <select name="normal_set" id="normal_set">
                            <option value="gencode_human_set" selected="selected">All Human lncRNA (Gencode)</option>
                            <option value="gencode_mouse_set">All Mouse lncRNA (Gencode)</option>
                            <option value="user_set">User Set</option>
                        </select>
                    </div>
                </div>

                <div class="row">

                    <div class="column">
                        <h2>Select kmer length</h2>
                    </div>

                    <div class="column">
                        <select id="kmer_length" name="kmer_length">
                            <option>1</option>
                            <option>2</option>
                            <option>3</option>
                            <option>4</option>
                            <option>5</option>
                            <option>6</option>
                            <option class="no_precompute">7</option>
                            <option class="no_precompute">8</option>
                            <option class="no_precompute">9</option>
                        </select>
                    </div>
                </div>
                <p id="kmer_warning" class="warning">Warning: This kmer length is too large to visualize. <br> K < 7 for visualization.</p>
            </div>
        </div>

        <div class="row break"></div>

        <div class="row">
            <div class="container single_column_small button">

                <input id="submit" type="submit" value="Submit">

            </div>
        </div>
    </div>

    <!--------------------------------------------------------->
    <!--Visualization Tab-->
    <!--------------------------------------------------------->
    <div id="visual" class="row">

        <div id="empty_message" class="single_column"><p>Please run SEEKR to populate this tab.</p></div>

        <div class="break"></div>

        <div id="warning_messages" class="single_column"></div>

        <!--div is hidden until algorithm is run-->
        <div id="results_toggle">

            <!--contains the tab buttons for displaying each visual-->
            <div class="row">
                <div class="single_column container">

                    <ul class="row">
                        <li class="tab sub_select_left"><a href="#pearson"><h2>Pearson Correlation Matrix (user set x comparison set)</h2></a></li>
                        <li class="tab tab2 sub_select_right"><a href="#kmer"><h2>Kmer Profiles Matrix (user set x kmer bin counts)</h2></a></li>
                    </ul>

                </div>
            </div>

            <!--Pearson Correlation visual-->
            <div id="pearson" class="row">

                <!--<div class="row">-->
                    <!--<div class="single_column">-->
                        <!--<h1>Pearson Correlations for User Set x Comparison Set</h1>-->
                    <!--</div>-->
                <!--</div>-->

                <!--Button tools for saving and re-ordering graph-->
                <div class="row">
                    <div class="single_column container">
                        <h3 class="visual_desc">Click on the axis labels to re-order the graph</h3>
                        <input id="pearson_save" class="tool" type="button" value="Save as .csv">
                        <input id="pearson_png" class="tool" type ="button" value="Save as .png">
                        <input id="pearson_reset" class="tool" type="button" value="Reset Ordering">
                    </div>
                </div>


                <div id="pearson_chart" class='canvas_container'></div>
            </div>

            <!--Kmer Profile visual-->
            <div id="kmer" class="row">

                <!--<div class="row">-->
                    <!--<div class="single_column">-->
                        <!--<h1>Normalized Kmer Counts for User Set</h1>-->
                    <!--</div>-->
                <!--</div>-->

                <!--Button tools for saving and reordering visual-->
                <div class="row">
                    <div class="single_column container">
                        <h3 class="visual_desc">Click on the axis labels to re-order the graph</h3>
                        <input id="kmer_save" class="tool" type="button" value="Save as .csv">
                        <input id="kmer_png" class="tool" type ="button" value="Save as .png">
                        <input id="kmer_reset" class="tool" type="button" value="Reset Ordering">
                    </div>
                </div>

                <div id="kmer_chart" class="canvas_container"></div>

            </div>

            <!--Toolstips are show using d3 to display hover info-->
            <div id="pearson_tooltip" class="hidden tooltip">
                <p><span id="p_value"></span></p>
            </div>

            <div id="kmer_tooltip" class="hidden tooltip">
                <p><span id="k_value"></span></p>
            </div>
        </div>

        <div id="nv_downloads" class="row">

            <div class="break"></div>

            <div class="row">
                <div class="single_column container">
                    <input id="nv_pearson_save" class="tool_download" type="button" value="Save Pearson Correlation Matrix as .csv">
                    <input id="nv_kmer_save" class="tool_download" type="button" value="Save Kmer Profiles Matrix as .csv">
                </div>
            </div>
        </div>
    </div>



    <!--------------------------------------------------------->
    <!--Help Page---------------------------------------------->
    <!--------------------------------------------------------->
    <div id="help" class="main">

        <div class="row">
            <div class="single_column container">

                <ul class="row">
                    <li class="tab sub_select_left"><a href="#about"><h2>About SEEKR</h2></a></li>
                    <li class="tab tab2 sub_select_right"><a href="#manual"><h2>User Manual</h2></a></li>
                </ul>

            </div>
        </div>



        <div id="about" class="row">
            <div class="single_column container">
                <p>
                    Welcome to the webapp interface for the SEEKR (SEquence Evaluation through Kmer Representation) algorithm.
                    SEEKR is a novel algorithm used to quantify similarities and differences in genomic sequences, particularly those of long non-coding RNAs (lncRNAs).
                    <br><br>
                    A longer description is coming soon!
                      <!--
                    The functions of most long non-coding RNAs (lncRNAs) are unknown.
                    In contrast to proteins, lncRNAs with similar function often lack linear sequence homology; thus, identification of function in one lncRNA rarely informs identification of function in others.
                    Reasoning that most lncRNAs act by recruiting protein intermediaries via short binding motifs, we hypothesized that similarity in motif content could identify lncRNAs of related function that lack linear similarity.
                    We developed a method, SEquence Evaluation through Kmer Representation (SEEKR), which deconstructs linear sequence relationships in lncRNAs to evaluate similarity based on abundance of short motifs called kmers.
                    -->
                    <br><br>
                    If you use SEEKR, please site <a>link</a>.
                </p>
            </div>
        </div>


        <div id="manual" class="row">

            <div class="single_column container">
                <p><br></p>
                <h1>Input</h1>
                    <p>SEEKR requires several inputs and options be set before running.</p>
                    <br>
                    <h2 class="link">1) User fasta file </h2>
                        <p>This file should contain the transcripts the user is interested in studying via SEEKR. The file must be in fasta format. See <a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi?CMD=Web&PAGE_TYPE=BlastDocs&DOC_TYPE=BlastHelp">https://blast.ncbi.nlm.nih.gov</a> for a definition of the fasta format. Additionally, we provide two sample fasta files for testing:
                        <br>
                        <br>
                        <a href="/static/sample_fasta_2.fa">example2.fa</a><br>
                        <a href="/static/sample_fasta_15.fa">example15.fa</a><br>

                        </p>
                    <h2 class="link">2) Comparison fasta file (default = User Set)</h2>
                        <p>A second fasta file must be declared to compare against the transcripts of the user fasta file. There are three types of options for this file:</p>
                        <ol>
                            <li>User Set – This will compare the User fasta file to itself.</li>
                            <li>Preloaded GENCODE fasta files – We provide the latest human and mouse lncRNA annotations from GENCODE.</li>
                            <li>Upload Set – The user can choose to upload a second fasta file for comparison. An example use case for uploading a second fasta file would be using a zebrafish lncRNA annotation if the transcripts in the `User fasta file` are all from the zebrafish genome.</li>
                        </ol>
                    <h2 class="link">3) Normalization Set (default = All Human lncRNAs (GENCODE)) </h2>
                        <p>After counting kmers for each sequence, SEEKR normalizes each kmer by subtracting by the mean and dividing by the standard deviation of each kmer. In many cases, it is reasonable to substitute these mean and standard deviation arrays with other, related values.</p>
                        <ol>
                            <li>All Human lncRNAs (GENCODE) – A common use case is when comparing a relatively small (< 100) lncRNAs of interest to each other. Because the set is small, extreme values are more likely to arise. It is often of interest to measure how a small set of lncRNAs compares to themselves, but when normalized to a full annotation of lncRNAs.</li>
                            <li>All Mouse lncRNAs (GENCODE) – This option is the same as above, but using mouse lncRNAs.</li>
                            <li>User Set – If studying a larger set of lncRNAs, such as a zebrafish annotation, select the user set for normalization.</li>
                        </ol>
                    <h2 class="link">4) Kmer length (default = 1)</h2>
                        <p>The size of the kmer to count in the fasta files. We recommend k=6 for most applications, though smaller kmers will process exponentially faster.</p>
                <p><br></p>
            </div>
            <div class="single_column container">
                <p><br></p>
                <h1>Output</h1>
                    <p>Press the <b>Submit</b> button and the algorithms output will be visualized in the ‘Results’ tab (files container over 200 RNA sequences and Kmers above 6 will not produce visuals)</p>
                    <br>
                    <h2 class="link">1) Sequence Kmer Profiles:</h2>
                        <p>Displays the kmer count profiles for sequences in the User Set. Each sequence is a row on the y axis, and each kmer is a column on the x axis. The sequences are automatically arranged according to a hierarchical clustering on the kmer contents of each sequence. Clicking on either the row or column labels will re-order the graph based on the values in the row or column clicked.</p>
                    <h2 class="link">2) Sequence Pearson Correlation Matrix</h2>
                        <p>Both the x axis and the y axis represent sequences from the User fasta file. Each element in the matrix indicates the Pearson Correlation R value between the two sequences in the given row and column. The sequences are automatically arranged according to the same hierarchical clustering as the Sequence Kmer Profiles visual. Clicking on either the row or column labels will re-order the graph based on the values in the row or column clicked.</p><br>
            </div>
        </div>
    </div>
</div>

<div class="single_column_small">
    <p id="loading">processing...</p>
</div>


<script type="text/javascript" src="{{ url_for('static', filename='visual.js') }}"></script>


</body>
</html>
